cmd/misc/main.go

```diff
@@ -1233,3 +1240,294 @@ OUTER:
+			return
+		}
+		if currEpoch > 0 { // guard against underflows
+			currEpoch = currEpoch - 1
+		}
+	} else {
+		_, lastEpoch := utils.GetFirstAndLastEpochForDay(endDay)
+		currEpoch = lastEpoch
+	}
+
+	lastPeriod := utils.SyncPeriodOfEpoch(uint64(currEpoch)) + 1 // we can look into the future
+
+	start := time.Now()
+	for p := firstPeriod; p <= lastPeriod; p++ {
+		t0 := time.Now()
+
+		err := reExportSyncCommittee(rpcClient, p, dryRun)
+		if err != nil {
+			if strings.Contains(err.Error(), "not found 404") {
+				logrus.WithField("period", p).Infof("reached max period, stopping")
+				break
+			} else {
+				logrus.WithError(err).WithField("period", p).Errorf("error re-exporting sync_committee")
+				return
+			}
+		}
+
+		logrus.WithFields(logrus.Fields{
+			"period":   p,
+			"epoch":    utils.FirstEpochOfSyncPeriod(p),
+			"duration": time.Since(t0),
+		}).Infof("re-exported sync_committee")
+	}
+
+	logrus.Infof("finished all exporting sync_committee for periods %v - %v, took %v", firstPeriod, lastPeriod, time.Since(start))
+}
+
+func exportSyncCommitteeValidatorStats(rpcClient rpc.Client, startDay, endDay uint64, dryRun, skipPhase1 bool) {
+	var lastEpoch = uint64(0)
+
+	if endDay <= 0 {
+		var err error
+		lastEpoch, err = db.GetLatestFinalizedEpoch()
+		if err != nil {
+			logrus.WithError(err).Errorf("error getting latest finalized epoch")
+			return
+		}
+		if lastEpoch > 0 { // guard against underflows
+			lastEpoch = lastEpoch - 1
+		}
+	} else {
+		_, lastEpoch = utils.GetFirstAndLastEpochForDay(endDay)
+	}
+
+	start := time.Now()
+
+	epochsPerDay := utils.EpochsPerDay()
+	if lastEpoch < epochsPerDay {
+		logrus.Infof("skipping exporting stats, first day has not been indexed yet")
+		return
+	}
+	currentDay := lastEpoch / epochsPerDay
+	previousDay := currentDay - 1
+
+	for day := startDay; day <= previousDay; day++ {
+		startDay := time.Now()
+		err := UpdateValidatorStatisticsSyncData(day, rpcClient, dryRun)
+		if err != nil {
+			utils.LogError(err, fmt.Errorf("error exporting stats for day %v", day), 0)
+			break
+		}
+
+		logrus.Infof("finished updating validators_stats for day %v, took %v", day, time.Since(startDay))
+	}
+
+	logrus.Infof("finished all exporting stats for days %v - %v, took %v", startDay, previousDay, time.Since(start))
+}
+
+func UpdateValidatorStatisticsSyncData(day uint64, client rpc.Client, dryRun bool) error {
+	exportStart := time.Now()
+	firstEpoch, lastEpoch := utils.GetFirstAndLastEpochForDay(day)
+
+	logrus.Infof("exporting statistics for day %v (epoch %v to %v)", day, firstEpoch, lastEpoch)
+
+	if err := db.CheckIfDayIsFinalized(day); err != nil && !dryRun {
+		return err
+	}
+
+	logrus.Infof("getting exported state for day %v", day)
+
+	var err error
+	maxValidatorIndex := uint64(1999999) //bt.GetMaxValidatorindexForEpoch(lastEpoch)
+	// if err != nil {
+	// 	logrus.Errorf("error getting max validator index for epoch %v: %v", lastEpoch, err)
+	// 	return err
+	// }
```

Comment:
Is this intentional or leftover debug code? IIRC `bt.GetMaxValidatorindexForEpoch` doesn't work for all networks, something like this would be safer:
https://github.com/manuelsc/eth2-beaconchain-explorer/blob/bf2fd38536d3615b5cac044512325cb361434b25/cmd/misc/main.go#L1195-L1196